packages/docs/src/components/DocSiteNavigation.tsx

```diff
@@ -123,57 +132,117 @@ const DocSiteNavigation = ({ location }: DocSiteNavProps) => {
-          defaultCollapsed: !isSubNavItemSelected(subNavItems),
-          selected: isSelected,
-        });
+      if (dataItem.label.split('/').length > 1) {
+        level2Items.push(dataItem);
       } else {
-        // for each level 1 item without sub nav items,
-        // add each top level item
-        dataItem.edges.forEach((navItemLvl1) => {
-          retVal.push(formatNavItemData(navItemLvl1.node));
-        });
+        level1Items[dataItem.label] = dataItem;
       }
     });
 
+    // iterate through level2 items and nest them under their level1 parent
+    level2Items.forEach((level2Item) => {
+      // sort level3 items by order then id / name
+      level2Item.items = sortBy(level2Item.items, ['order', 'id']);
+
+      const [level1Name, level2Name] = level2Item.label.split('/');
+
+      level1Items[level1Name].items.push({ ...level2Item, label: level2Name });
+
+      if (!level2Item.defaultCollapsed) {
+        // if any level2 item is default expanded, make sure that level1 item also gets default expanded
+        level1Items[level1Name].defaultCollapsed = level2Item.defaultCollapsed;
+      }
+    });
+
+    return Object.values(level1Items);
+  };
+
+  /**
+   * sort items - needed because level2 items that have sub nav are added to the end of list in previous step
+   */
+  const sortNavItems = (level1Items: VerticalNavItemProps[]): VerticalNavItemProps[] => {
+    level1Items.forEach((level1Item) => {
+      // sort items based on the order defined in frontmatter then id which is the file path for the item
+      level1Item.items = sortBy(level1Item.items, ['order', 'id']);
+    });
+
+    level1Items.sort((itemA, itemB) => {
+      return level1ItemOrder.indexOf(itemA.label) - level1ItemOrder.indexOf(itemB.label);
+    });
+    return level1Items;
+  };
+
+  /**
+   * re-structures graphQL object to VerticalNavItem structure
+   */
+  const restructureAsVerticalNavItem = (dataList): VerticalNavItemProps[] => {
+    const retVal: VerticalNavItemProps[] = [];
+    dataList.forEach((dataItem) => {
+      // format all the level 2 items
+      const subNavItems = dataItem.edges.map((subNavItem) => formatNavItemData(subNavItem.node));
+      // in order to sort properly, need to get the smallest leve l3 order value and assign to level 2
+      const subNavItemsOrderVals = subNavItems.map((subNavItem) => subNavItem.order);
+      const orderVal = Math.min(...subNavItemsOrderVals);
+
+      const labelText = formatNavItemLabel(dataItem.fieldValue);
+      const isSelected = isItemSelected(dataItem.fieldValue);
+
+      // add level 1 item & sub items
+      retVal.push({
+        label: labelText,
+        items: subNavItems,
+        defaultCollapsed: !isSubNavItemSelected(subNavItems),
+        selected: isSelected,
+        id: dataItem.fieldValue,
+        order: orderVal,
+      });
+    });
+
     return retVal;
   };
 
+  const formatNavData = (graphQlData: GraphQlNavItemNode[]): VerticalNavItemProps[] => {
```

Comment:
These functions are now substantial enough to consider moving out of the component. It looks like they get all the data they need through parameters.

Reply:
Done! Moved in a utils file